thirdparty/github.go

```diff
@@ -1765,6 +1765,104 @@ func GetMergeablePullRequest(ctx context.Context, issue int, githubToken, owner,
 	return pr, nil
 }
 
+// HasEvergreenBranchProtections checks if any branch protection rule begins "evergreen".
+// We deliberately check for the prefix because users may have installed variant-level branch protections.
+func HasEvergreenBranchProtection(ctx context.Context, token, owner, repo, branch string) (bool, error) {
```

Comment:
I'm seeing you already wrote [GetEvergreenBranchProtection](https://github.com/evergreen-ci/evergreen/blob/5df5c07ddd9bd75953acdefd3c9b6acaabeee319/thirdparty/github.go#L2016-L2015) -- can we just adapt that one?

Reply:
I have a mild preference for this just returning the branch protection so it can also just be used in the install function( if we keep it) as opposed to duplicating that logic

Reply:
I removed the duplicate logic. I suspect we do want a function that returns a boolean. This now just wraps the other function.

Reply:
okay sorry to keep commenting on https://github.com/evergreen-ci/evergreen/pull/7012 i think i’m not explaining myself well — the thing i’m trying to do away with is that both the Install and the Get function set up and call GetBranchProtection https://github.com/evergreen-ci/evergreen/blob/5df5c07ddd9bd75953acdefd3c9b6acaabeee319/thirdparty/github.go#L2035 

I feel like ideally this code would be written only once and would just return the result of GetBranchProtection (or return all of the checks at least) so then both the Install and Get functions would be able to use the result. I'm going to go ahead and approve though since the gist works.